code/game/machinery/engi_points_vendor.dm

```diff
@@ -0,0 +1,183 @@
+	icon_state = "store"
+	verb_say = "states"
+	density = 1
+	anchored = 1
+	var/obj/item/device/radio/radio
+	var/GBP = 0
+	var/GBPearned = 0
+	var/power_export_bonus = 0
+	var/air_alarm_bonus = 0
+	var/power_alarm_bonus = 0
+	var/fire_alarm_bonus = 0
+	var/alarm_rating = ""
+	var/prior_bonus = 2500
+	var/total_bonus = 0
+	var/GBP_alarm_cooldown = 4500
+	var/list/prize_list = list(
+		new /datum/GBP_equipment("Tendie",				/obj/item/weapon/reagent_containers/food/snacks/nugget,				50,		1),
+		new /datum/GBP_equipment("Cigar",				/obj/item/clothing/mask/cigarette/cigar/havana,						50,		1),
+		new /datum/GBP_equipment("Soap",				/obj/item/weapon/soap/nanotrasen,									100,	1),
+		new /datum/GBP_equipment("Fulton Beacon",		/obj/item/fulton_core,												50,		1),
+		new /datum/GBP_equipment("Fulton Pack",			/obj/item/weapon/extraction_pack,									200,	1),
+		new /datum/GBP_equipment("Space Cash",			/obj/item/stack/spacecash/c1000,									250,	1),
+		new /datum/GBP_equipment("Insulated Gloves",				/obj/item/clothing/gloves/color/yellow,					400,	1),
+		new /datum/GBP_equipment("50 metal sheets",			/obj/item/stack/sheet/metal/fifty,								500,	1),
+		new /datum/GBP_equipment("50 glass sheets",			/obj/item/stack/sheet/glass/fifty,								500,	1),
+		new /datum/GBP_equipment("50 cardboard sheets",			/obj/item/stack/sheet/cardboard/fifty,						500,	1),
+		new /datum/GBP_equipment("Hardsuit x3",			/obj/item/clothing/suit/space/hardsuit,								750,	3),
+		new /datum/GBP_equipment("Jetpack Upgrade x5",		/obj/item/weapon/tank/jetpack/suit,								1000,	5),
+		new /datum/GBP_equipment("Powertools x3",			/obj/item/weapon/storage/belt/utility/chief/full,				2000,	3),
+		new /datum/GBP_equipment("Advanced Magboot x3",			/obj/item/clothing/shoes/magboots/advance,					3000,	3),
+		new /datum/GBP_equipment("Reflector Box x3",			/obj/structure/reflector/box,								3500,	3),
+		new /datum/GBP_equipment("Radiation Collector x3",			/obj/machinery/power/rad_collector,						4000,	3),
+		new /datum/GBP_equipment("Ranged RCD x3",			/obj/item/weapon/rcd/arcd,										6000,	3),
+		new /datum/GBP_equipment("ERT Hardsuit x5",		/obj/item/clothing/suit/space/hardsuit/ert/engi,					7000,	5),
+		new /datum/GBP_equipment("Portal Gun x5",			/obj/item/weapon/gun/energy/wormhole_projector,					8000,	5),
+		new /datum/GBP_equipment("Prototype Repair Vehicle x2",		/obj/vehicle/space/speedbike/repair,					10000,	2),
+		new /datum/GBP_equipment("Reactive Decoy Armor x5",		/obj/item/clothing/suit/armor/reactive/stealth,				12500,	5),
+		new /datum/GBP_equipment("Chrono Suit x5",		/obj/item/clothing/suit/space/chronos,								20000,	5),
+		new /datum/GBP_equipment("WHAT HAVE YOU DONE... x5",		/obj/vehicle/space/speedbike/memewagon,				30000,	5),
+		)
+
+/datum/GBP_equipment
+	var/equipment_name = "generic"
+	var/equipment_path = null
+	var/cost = 0
+	var/amount = 0
+
+/datum/GBP_equipment/New(name, path, cost, amount)
+	src.equipment_name = name

```

Comment:
ok this one is almost certainly copied from the mining vendor but im too lazy to check, chances are almost any weird source usage is pulled from somewhere.